src/t_set.c

```diff
@@ -1014,7 +1014,7 @@ void sinterCommand(client *c) {
 void sinterCardCommand(client *c) {
     long j;
     long numkeys = 0; /* Number of keys. */
-    long limit = 0;   /* 0 means not limit. */
+    long limit = -1;  /* 0 means not limit. */
```

Comment:
Maybe we can assume that limit has to be larger than 0 (can't be equal to 0). And then we don't need to fix it to 0 if it remains -1 after the arg parsing?

Reply:
> This PR also fixes the problem of passing in multiple identical
parameters and adds some tests to cover it.

actuall looking at it now, i want to remove this part of the changes... (it doesn't feel particularly necessary, there are other places that exist forever. And this part make the diff become bigger)

> can't be equal to 0

Yes, we can. but we allow explicit passing `LIMIT 0` (in SINTERCARD or ZINTERCARD)

Reply:
I see, so if we want to detect someone passed a `LIMIT` arg we have to have some unique value. I guess this is ok.